### ISSUE TITLE
Improve UX for `to_unit` and `to(unit=...)`?

### ISSUE THREAD

**Issue by SimonHeybrock** [scipp/scipp#3029]
Currently, we have to do this in quite a few places:

```python
var = var.to(unit=other.unit)
```
or worse, if `other` may contain binned data:
```python
def elem_unit(var: VariableLike) -> sc.Unit:
    return var.bins.unit if var.bins is not None else var.unit

var = var.to(unit=elem_unit(other))
```

We may consider supporting passing another scipp object, which unit we want to match instead:

```python
var = var.to(unit=other)  # match var.bins.unit or var.unit
```

There may be concerns about ambiguity. I am not sure there are any, but nevertheless, another option would be to provide a named function (or method), or different keyword arg:

```python
var = sc.match_unit(var, other)
# or
var = var.to(match_unit=other)
```

The latter could be extended to the `dtype` as well:

```python
var1 = var.to(match_unit=other)
var2 = var.to(match_dtype=other)
var3 = var.to(match=other)
```

**Comment by jl-wynen** [comment]
I don't think this can be ambiguous for `unit`. But it can be for `dtype`. However, converting to/from `Variable` or `DataArray` is not supported anyway.

But it is ambiguous with binned data. Should it use the `elem_unit` and `elem_dtype` or the outer one? Probably the `elem_*` in the majority of cases. But is this good enough to make that the default?

**Comment by SimonHeybrock** [comment]
Binned data cannot have an outer unit (i.e., it is always `unit=None`), so it cannot be ambiguous there either.
